server/backend/api/app/schemas/pagination.py

```diff
@@ -0,0 +1,66 @@
+
+from abc import ABC
+from typing import Generic, List
+
+from api.app.constants import (DEFAULT_PAGE_SIZE, MAX_PAGE_SIZE, MIN_PAGE,
+                               MIN_PAGE_SIZE, SEARCH_FIELD_MAX_LENGTH,
+                               SEARCH_FIELD_MIN_LENGTH, SortOrderEnum, T,
+                               UserRoleSortByEnum)
+from fastapi import Query
+from pydantic import BaseModel, Field
+
+"""
+Schema objects for the backend pagination, sorting, filtering related purpose.
+"""
+
+"""
+Note: FastAPI does not seem to work well with Pydantic for Query Parameter + Swagger.
+To use Pydantic class in "router's endpoint" as FastAPI query parameter, it needs to wrap the property with "Field(Query(...))"
+like below example in order to get validation correct and with 'description' shows up at Swagger:
+    page_number: int | None = Field(Query(default=1, ge=1, description="page number to get the paged data"))
+
+More over, the endpoint also need to use "Depends()" for function argument, like below exampe:
+    def get_fam_application_user_role_assignment(
+        application_id: int,
+        page_params: PageParamsSchema = Depends(),
+        ...
+    )
+
+And, in this case, the validation error default from Pydantic will not be 400, it will be 422 (from Pydantic)
```

Comment:
Note, currently built in validation errors from Pydantic is 422. Hope that's acceptable for frontend.